src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -536,8 +536,10 @@ trait Namers extends MethodSynthesis {
       val Import(expr, selectors) = tree
       val base = expr.tpe
 
-      def checkNotRedundant(pos: Position, from: Name, to0: Name) {
-        def check(to: Name) = {
+      // warn proactively if specific import loses to definition in scope,
+      // since it may result in desired implicit not imported into scope.
+      def checkNotRedundant(pos: Position, from: Name, to0: Name): Unit = {
+        def check(to: Name): Unit = {
```

Comment:
Um. I guess since we're talking import magic. I know the check used to be wrong, false positive. Now I'm not sure what the user story is. Oh, this is covered by unused import. So is this check only guarding against old bugs in name binding? Including implicit shadowing?
```
object X {
  val global = 0
  import concurrent.ExecutionContext.global
  println(global)
}
```

Reply:
Yes, I see bug 2866 is due to Jason: "I have five unanswered questions"... after grepping for permanently hidden again in the check files...

Reply:
Yeah, I dropped your changes to this code from #5220 as I didn't understand the original code or the change. 🤷‍♂

Reply:
I'm not sure it's very meaningful, even if it had a bug-compatible meaning once.